### PR TITLE
Fix usage of NuGet.config that was missed in an earlier commit

### LINKED ISSUE #2747: Need to handle multiple cases of "NuGet.config".
I did a partial fix to updating DNX to understand the new NuGet.config (required by nuget.exe v3). You typically won't see issues until you go cross plat where file systems are case sensitive.

For context on this issue see the comment I left on the [commit](https://github.com/aspnet/dnx/commit/4cb916d2e0f88d6a62fa5224dbb5e65e77523068).

### PR DESCRIPTION
**Pull request by pranavkm**
Additionally remove unused code from Settings.

Fixes #2747

### CODE REVIEW

**Comment by pranavkm** [comment]
Would be useful to look at individual commits.

**Comment by anurse** [comment]
This is probably going to be unnecessary after my work on #1847 which is going to replace our custom code with NuGet.Configuration.

**Comment by pranavkm** [comment]
We'll need to port the multiple path support back to NuGet. Right now, they only support `NuGet.config`

**Comment by davidfowl** [comment]
Why is travis failing

**Comment by pranavkm** [comment]
@davidfowl, no idea:

```
=== TEST EXECUTION SUMMARY ===
   Microsoft.Dnx.Tooling.Tests  Total: 23, Errors: 0, Failed: 0, Skipped: 0, Time: 4.799s

No output has been received in the last 10 minutes, this potentially indicates a stalled build or something wrong with the build itself.
The build has been terminated
```

**Comment by anurse** [comment]
We've seen a few of these in random repositories at random times. Xunit issue maybe?

File: src/Microsoft.Dnx.Tooling/NuGet/Core/Configuration/Settings.cs [hunk 1]
```diff
@@ -174,8 +174,16 @@ public string ConfigFilePath
...
                 var userSettingsDir = DnuEnvironment.GetFolderPath(DnuFolderPath.UserSettingsDirectory);
-                appDataSettings = ReadSettings(new PhysicalFileSystem(userSettingsDir),
-                    Constants.SettingsFileName);
+                var fileName = SettingsFileNames
+                    .Select(settingsFileName => Path.Combine(userSettingsDir, settingsFileName))
+                    .FirstOrDefault(fileSystem.FileExists);
+
+                if (fileName != null)
+                {
+                    appDataSettings = ReadSettings(
```
Comment (troydai) [inline src/Microsoft.Dnx.Tooling/NuGet/Core/Configuration/Settings.cs]: Get this on the same line?

**Comment by troydai** [comment]
:shipit:

**Comment by davidfowl** [comment]
This breaks things. Do over

**Comment by pranavkm** [comment]
Updated

**Comment by davidfowl** [comment]
Travis

**Comment by davidfowl** [comment]
:shipit:

**Pull request closed without merging**